### PR TITLE
PPTP-868 : Generic retry functionality available via Retry trait. Non…

### PR DESCRIPTION
**Pull request by RyanKellyHMRC**
… Repudiation Service submission call will be retried (as per configuration) on 500 errors.

### CODE REVIEW

**Comment by PauloPortugal** [comment]
Quick commit review.
There seems to be a lot of detail on the commit subject.
My suggestion is to use the commit body for implementation details or explain _why_, _how_ and _what_.


```
PPTP-868 Generic retry functionality available via Retry trait.

Non Repudiation Service submission call will be retried (as per configuration) on 500 errors.
```

We should loosely follow https://chris.beams.io/posts/git-commit/#seven-rules

**Comment by RyanKellyHMRC** [comment]
> Quick commit review.
> There seems to be a lot of detail on the commit subject.
> My suggestion is to use the commit body for implementation details or explain _why_, _how_ and _what_.
> 
> ```
> PPTP-868 Generic retry functionality available via Retry trait.
> 
> Non Repudiation Service submission call will be retried (as per configuration) on 500 errors.
> ```
> 
> We should loosely follow https://chris.beams.io/posts/git-commit/#seven-rules
> 
> how to do this on the command line:
> 
> ```
> git commit --amend 
> ```

Apologies @PauloPortugal; I remember you specifically asking for this. Old habits die hard.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/config/AppConfig.scala [hunk 1]
```diff
@@ -45,4 +47,5 @@ class AppConfig @Inject() (config: Configuration, servicesConfig: ServicesConfig
   lazy val nonRepudiationApiKey: String =
     servicesConfig.getString("microservice.services.nrs.api-key")
 
+  val nrsRetries: Seq[FiniteDuration] = config.get[Seq[FiniteDuration]]("nrs.retries")
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/config/AppConfig.scala:50]: :+1:

**Review by PauloPortugal — APPROVED** [review]
LGTM

**Pull request merged**